package/MDAnalysis/core/topologyattrs.py

```diff
@@ -2038,8 +2074,412 @@ def total_charge(group, compound='group'):
+            "mass" or the center of "charge", default is "mass".
+
+        Returns
+        -------
+        numpy.ndarray
+            Dipole vector(s) of (compounds of) the group in :math:`eÅ`.
+            If `compound` was set to ``'group'``, the output will be a single
+            value. Otherwise, the output will be a 1d array of shape ``(n,3)``
+            where ``n`` is the number of compounds.
+
+
+        .. versionadded:: 2.4.0
+        """
+
+        compound = compound.lower()
+
+        atomgroup = group.atoms
+        charges = atomgroup.charges
+
+        if center == "mass":
+            masses = atomgroup.masses
+            ref = atomgroup.center_of_mass(wrap=wrap,
+                                           unwrap=unwrap,
+                                           compound=compound)
+        elif center == "charge":
+            ref = atomgroup.center_of_charge(wrap=wrap,
+                                             unwrap=unwrap,
+                                             compound=compound)
+        else:
+            choices = ["mass", "charge"]
+            raise ValueError(
+                f"The dipole center, {center}, is not supported. Choose"
+                " one of: {choices}")
+
+        if compound == 'group':
+            if wrap:
+                recenteredpos = (atomgroup.pack_into_box(inplace=False) - ref)
+            elif unwrap:
+                recenteredpos = (atomgroup.unwrap(
+                    inplace=False,
+                    compound=compound,
+                    reference=None,
+                ) - ref)
+            else:
+                recenteredpos = (atomgroup.positions - ref)
+            dipole_vector = np.sum(recenteredpos * charges[:, np.newaxis],
+                                   axis=0)
+        else:
+            (atom_masks, compound_masks,
+             n_compounds) = atomgroup._split_by_compound_indices(compound)
+
+            if unwrap:
+                coords = atomgroup.unwrap(compound=compound,
+                                          reference=None,
+                                          inplace=False)
+            else:
+                coords = atomgroup.positions
+            chgs = atomgroup.charges
+
+            dipole_vector = np.empty((n_compounds, 3), dtype=np.float64)
+            for compound_mask, atom_mask in zip(compound_masks, atom_masks):
+                dipole_vector[compound_mask] = np.sum(
+                    (coords[atom_mask] - ref[compound_mask][:, None, :]) *
+                    chgs[atom_mask][:, :, None],
+                    axis=1)
+
+        return dipole_vector
+
+    transplants[GroupBase].append(('dipole_vector', dipole_vector))
+
+    def dipole_moment(group, **kwargs):
+        r"""Dipole moment of the group or compounds in a group.
+
+        .. math::
+            \mu = |\boldsymbol{\mu}| = \sqrt{ \sum_{i=1}^{D} \mu^2 }
+
+        Where :math:`D` is the number of dimensions.
```

Comment:
We always compute 3D? There is no dimension argument so perhaps make that clear.